core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-page-dropzone/ema-page-dropzone.component.ts

```diff
@@ -155,11 +91,35 @@ export class EmaPageDropzoneComponent {
      * @return {*}  {boolean}
      * @memberof EmaPageDropzoneComponent
      */
-    private isTop(event: DragEvent): boolean {
+    private calculatePosition(event: DragEvent): void {
         const target = event.target as HTMLDivElement;
+        const parentRect = this.el.nativeElement.getBoundingClientRect();
         const targetRect = target.getBoundingClientRect();
         const mouseY = event.clientY;
+        const isTop = mouseY < targetRect.top + targetRect.height / 2;
 
-        return mouseY < targetRect.top + targetRect.height / 2;
+        this.position.set(isTop ? 'before' : 'after');
+        this.parentRect.set(parentRect);
+        this.targetRect.set(targetRect);
```

Comment:
Do we need them separately? Or can we unify them in a single Signal?

Reply:
I can use one signal indeed.

Reply:
Done!